solver/src/solver/oneinch_solver.rs

```diff
@@ -93,12 +97,10 @@ where
         );
 
         let spender = self.client.get_spender().await?;
-        let sell_token = ERC20::at(&self.web3(), order.sell_token);
-
         // Fetching allowance before making the SwapQuery so that the Swap info is as recent as possible
-        let existing_allowance = self
+        let approval = self
```

Comment:
Might be nice to have a caching layer here as well.

Reply:
The difficulty with caching is that its quite hard to determine when to invalidate an entry.

I do like the suggestion though - I would just have to think a bit about how to do it.